lib/src/__tests__/DatePickerTestingLib.test.tsx

```diff
@@ -78,4 +78,24 @@ describe('<DatePicker />', () => {
 
     expect(screen.queryByRole('dialog')).toBeInTheDocument();
   });
+
+  it('Does not call onChange if same date selected', async () => {
+    const onChangeMock = jest.fn();
+
+    render(
+      <DesktopDatePicker
+        TransitionComponent={FakeTransitionComponent}
+        disableCloseOnSelect
+        value={utilsToUse.date('2018-01-01T00:00:00.000Z')}
+        onChange={onChangeMock}
+        renderInput={props => <TextField {...props} />}
+      />
+    );
+
+    fireEvent.click(screen.getByLabelText('Choose date, selected date is Jan 1, 2018'));
```

Comment:
I side note, I wonder if the asserting here isn't too brittle. What happens if we change the formation or the value? Would it be possible to query an element with role button instead?

Reply:
I too think that this approach to querying can be brittle. But also fixing it should probably part of a separate issue/pull request?

Reply:
Yeah, right, a different pull request would make sense